sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/EventHubReactorAmqpConnection.java

```diff
@@ -87,9 +89,15 @@ public EventHubReactorAmqpConnection(String connectionId, ConnectionOptions conn
                 "connectionId[%s]: Connection is disposed. Cannot get management instance", connectionId))));
         }
 
-        return Mono.defer(() -> getReactorConnection().thenReturn(new ManagementChannel(
-                createRequestResponseChannel(MANAGEMENT_SESSION_NAME, MANAGEMENT_LINK_NAME, MANAGEMENT_ADDRESS),
-                eventHubName, tokenCredential, tokenManagerProvider, this.messageSerializer, scheduler)));
+        return getReactorConnection().then(Mono.fromCallable(() -> {
+            if (managementChannel == null) {
```

Comment:
This should be synchronized or declare `this.managementChannel` as AtomicReference.